src/text-editor-presenter.coffee

```diff
@@ -94,40 +82,68 @@ class TextEditorPresenter
     @updateCustomGutterDecorationState() if @shouldUpdateCustomGutterDecorationState
     @updating = false
 
+    @resetTrackedUpdates()
+
     @state
 
+  resetTrackedUpdates: ->
+    @shouldUpdateFocusedState = false
+    @shouldUpdateHeightState = false
+    @shouldUpdateVerticalScrollState = false
+    @shouldUpdateHorizontalScrollState = false
+    @shouldUpdateScrollbarsState = false
+    @shouldUpdateHiddenInputState = false
+    @shouldUpdateContentState = false
+    @shouldUpdateDecorations = false
+    @shouldUpdateLinesState = false
+    @shouldUpdateCursorsState = false
+    @shouldUpdateOverlaysState = false
+    @shouldUpdateLineNumberGutterState = false
+    @shouldUpdateLineNumbersState = false
+    @shouldUpdateGutterOrderState = false
+    @shouldUpdateCustomGutterDecorationState = false
+
   observeModel: ->
     @disposables.add @model.onDidChange =>
       @updateContentDimensions()
       @updateEndRow()
-      @updateHeightState()
-      @updateVerticalScrollState()
-      @updateHorizontalScrollState()
-      @updateScrollbarsState()
-      @updateContentState()
-      @updateDecorations()
-      @updateLinesState()
-      @updateLineNumberGutterState()
-      @updateLineNumbersState()
-      @updateGutterOrderState()
-      @updateCustomGutterDecorationState()
+      @shouldUpdateHeightState = true
```

Comment:
Totally. Credits to @nathansobo :dragon: for suggesting this refactoring! :sparkles: 

On a side note, I have noticed we handle *batched operations* in more than one object. Lately I have been thinking about some abstraction which could handle batching for us through a more descriptive API:

```coffee
# Record actions to execute later
foo.defer(@updateHeightState) # or foo.push(@updateHeightState)
foo.defer(@updateWidthState)
foo.defer(@updateBar)

# Apply recorded operations just-in-time (each "operation type" should be executed once and only once, or we could have several strategies to handle that if we need to)
foo.apply()
```

Where `foo` could be `LazyWorkQueue`, or something similar.

Anyway, booleans already express the intent much better than the previous implementation, so I am also fine to keep this stuff as it is for the time being :+1: